src/Compiler/perf/Microsoft.AspNetCore.Razor.Microbenchmarks.Generator/Benchmarks.cs

```diff
@@ -0,0 +1,148 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using BenchmarkDotNet.Attributes;
+using Microsoft.Build.Locator;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.AspNetCore.Razor.Microbenchmarks.Generator;
+
+public class Benchmarks
+{
+    ProjectSetup.RazorProject? _project;
+
+    public enum ChangeKind { Independent, DependentIgnorable, Dependent };
+    [ParamsAllExceptDebug(ChangeKind.Independent)]
+    public ChangeKind Change { get; set; }
+
+    public enum StartupKind { Warm, Cold };
+    [ParamsAllExceptDebug(StartupKind.Warm)]
+    public StartupKind Startup { get; set; }
+
+    [ModuleInitializer]
+    public static void LoadMSBuild() => MSBuildLocator.RegisterDefaults();
```

Comment:
Can we avoid using MSBuildWorkspace for our benchmarks? I'd much rather use an AdhocWorkspace and Jared's assembly references nugets to more precisely control the environment.

Reply:
Yeah, I think that's probably a good idea. I'm going to punt it to a future PR though as its working for now.

Created https://github.com/dotnet/razor/issues/7982 to track